### ISSUE TITLE
[BUG] - Unable to use multiple capsuleUserGroups | Important

### ISSUE THREAD

**Issue by ArunSisodiya** [clastix/capsule-proxy#143]
# BUG
Hi @prometherion / Capsule Team,

We have recently started using capsule in our organization to provide a multi-tenancy environment but we are facing some issues while using multiple capsuleUserGroups in `capsule-proxy`

As we have integrated `OIDC` for authentication and authorization, we are using capsule-proxy and doing the authorization based on `Groups` to which the user belongs.

As per the documentation - https://github.com/clastix/capsule-proxy/tree/master/charts/capsule-proxy, we can use multiple `capsuleUserGroups` in capsule-proxy but even after using this option capsule proxy is loading the last group mentioned in the list.

This is how I am defining the user groups -

```yaml
values:
    options:
      capsuleUserGroups: [ "group1", "group2", "group3" ]
```

My CapsuleConfiguraion look like this -

```yaml
apiVersion: capsule.clastix.io/v1alpha1
kind: CapsuleConfiguration
metadata:
  annotations:
    meta.helm.sh/release-name: capsule
    meta.helm.sh/release-namespace: capsule-system
  creationTimestamp: "2021-09-16T20:12:41Z"
  generation: 7
  labels:
    app.kubernetes.io/instance: capsule
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: capsule
    app.kubernetes.io/version: 0.1.0
    helm.sh/chart: capsule-0.1.0
    helm.toolkit.fluxcd.io/name: capsule
    helm.toolkit.fluxcd.io/namespace: capsule-system
  name: default
  resourceVersion: "902835"
  uid: ad34fba7-4b70-4309-addc-e041b90634fd
spec:
  forceTenantPrefix: false
  protectedNamespaceRegex: ""
  userGroups:
  - group1
  - group2
  - group3

```

And here are the logs from my capsule-proxy -

```bash
{"level":"info","ts":"2021-09-17T11:33:39.529Z","logger":"main","msg":"---"}
{"level":"info","ts":"2021-09-17T11:33:39.628Z","logger":"main","msg":"Manager listening on port 9001"}
{"level":"info","ts":"2021-09-17T11:33:39.628Z","logger":"main","msg":"Listening on HTTPS: false"}
{"level":"info","ts":"2021-09-17T11:33:39.628Z","logger":"main","msg":"The selected Capsule User Group is group3"}
{"level":"info","ts":"2021-09-17T11:33:39.628Z","logger":"main","msg":"The OIDC username selected is email"}
{"level":"info","ts":"2021-09-17T11:33:39.628Z","logger":"main","msg":"---"}
{"level":"info","ts":"2021-09-17T11:33:39.628Z","logger":"main","msg":"Creating the manager"}
I0917 11:33:41.728755       1 request.go:645] Throttling request took 1.099615878s, request: GET:https://<ip>:443/apis/route53.aws.crossplane.io/v1alpha1?timeout=32s
{"level":"info","ts":"2021-09-17T11:33:43.435Z","logger":"controller-runtime.metrics","msg":"metrics server is starting to listen","addr":":8080"}
{"level":"info","ts":"2021-09-17T11:33:43.728Z","logger":"main","msg":"Creating the Field Indexer"}
{"level":"info","ts":"2021-09-17T11:33:43.828Z","logger":"main","msg":"Creating the NamespaceFilter runner"}
{"level":"info","ts":"2021-09-17T11:33:44.028Z","logger":"main","msg":"Adding the NamespaceFilter runner to the Manager"}
{"level":"info","ts":"2021-09-17T11:33:44.028Z","logger":"main","msg":"Starting the Manager"}
{"level":"info","ts":"2021-09-17T11:33:44.629Z","logger":"controller-runtime.manager","msg":"starting metrics server","path":"/metrics"}
```

From my observation, I can see that `capsule-proxy` is picking some old codebase as I have seen that `The selected Capsule User Group is` log message has been changed to `The selected Capsule User Groups are`. 

Can you please help me to look into this issue on an urgent basis as we are blocked due to this and cannot onboard new tenants?

Please let me know if you need any information from my end.

**Comment by MaxFedotov** [comment]
Hi @ArunSisodiya.
Can you please share which versions of `capsule-proxy` and `capsule` you are using in your environment?

**Comment by ArunSisodiya** [comment]
Hi @MaxFedotov ,

```bash
capsule                    	capsule-system	3       	2021-09-17 11:09:43.731018483 +0000 UTC	deployed	capsule-0.1.0                    	0.1.0
capsule-proxy              	capsule-system	1       	2021-09-17 13:31:47.848893 +0200 CEST  	deployed	capsule-proxy-0.1.4              	0.1.0
```

Here is an overview of the capsule and capsule-proxy helm releases.

**Comment by prometherion** [comment]
👋🏻  @ArunSisodiya and thanks for the bug report!

Feature #133 should address your issue, it has been merged in master and we're planning the release of Capsule Proxy v0.1.1 containing several enhancements and bug fixes.

If you don't mind, I'd like to ask you to build `capsule-proxy` from the `master` branch and give it a try in your environment passing multiple user groups via CLI.

**Comment by ArunSisodiya** [comment]
@prometherion - Thanks for the reply. I thought like I am doing something wrong :)
So do you have any timelines, when you guys are planning to release v0.1.1? 

Then I will plan things accordingly at my end. :)

**Comment by prometherion** [comment]
> when you guys are planning to release v0.1.1?

According to the [v0.1.1 milestone](https://github.com/clastix/capsule-proxy/milestone/3), we're ready for the delivery, just the changelog must be put in place, so I'd say next week 🚀

**Comment by prometherion** [comment]
Well, I was thinking that, in the end, would be great to extract the Capsule user group array from the `CapsuleConfiguration` CR rather than repeating the values every time.

With that, all the changes would be reflected on `capsule-proxy` without the need of restarting the application.

Referencing this in a new issue so we can track it down.

**Comment by prometherion** [comment]
@ArunSisodiya this has been fixed with v0.1.1 release, please, open it back in case the problem persists.
